File: 剑指offer/45/main.cpp
class Solution {
public:
    string minNumber(vector<int>& nums) {
        string ans = "";
        // 比较函数使用 lambda 表达式
        sort(nums.begin(), nums.end(), [](const int& a, const int& b) {
            string n = to_string(a), m = to_string(b);
            // 直接使用 int 拼接可能会溢出，所以转换成字符串
            // 转换成字符串比较
            return m + n > n + m;
        });

        for (int num: nums)
            ans += to_string(num);
            
        return ans;
    }
};

File: 剑指offer/45/main.js
/**
 * @param {number[]} nums
 * @return {string}
 */
var minNumber = function(nums) {
  // 比较函数的返回值需要查询 mdn
  // https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/sort
  return nums.sort((n, m) => {
      n = n.toString()
      m = m.toString()
      if (m + n < n + m) {
          return 1
      } else if (m + n > n + m) {
          return -1
      } else {
          return 0
      }
  }).join('')
};